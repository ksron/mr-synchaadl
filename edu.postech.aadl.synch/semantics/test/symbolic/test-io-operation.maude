
set show timing off .

load ../../interpreter-symbolic.maude

load module-ba-semantics

---------------------------------------------------
select TEST-FEATURE-OPERATION .

red readFeature(
    < a :  DataInPort | content : (r(0) # bool(false)), cache : (r(1) # bool(false)) >
    < b :  DataInPort | content : (r(0) # bool(true)),   cache : (r(1) # bool(true))  >
    < c :  DataInPort | content : (r(0) # b(0)),    cache : (r(0) # b(1))    >
    < d : DataOutPort | content : (* # b(0)) >
    < e : DataOutPort | content : (real(0) # bool(false)) >
  ) .

red writeFeature(
    (d |-> (* # bool(true)), e |-> (r(1) # b(1)))
,
    < a :  DataInPort | content : (r(0) # bool(false)), cache : (r(1) # bool(false)) >
    < b :  DataInPort | content : (r(0) # bool(true)),  cache : (r(1) # bool(true))  >
    < c :  DataInPort | content : (r(0) # b(0)),    cache : (r(0) # b(1))    >
    < d : DataOutPort | content : (* # b(0)) >
    < e : DataOutPort | content : (real(0) # bool(false)) >
  ) .


---------------------------------------------------
select TEST-DATA-OPERATION .

red readData(
    < c1 : Data | value : real(3/4) # b(1), features : none, subcomponents : none, connections : empty, properties : none >
    < c2 : Data | value : r(0) # bool(true), features : none, subcomponents : none, connections : empty, properties : none >) .

red writeData((c1 |=> real(0) # bool(true), c2 |=> real(1) # b(1)),
    < c1 : Data | value : real(3/4) # b(1), features : none, subcomponents : none, connections : empty, properties : none >
    < c2 : Data | value : r(0) # bool(true), features : none, subcomponents : none, connections : empty, properties : none >) .


