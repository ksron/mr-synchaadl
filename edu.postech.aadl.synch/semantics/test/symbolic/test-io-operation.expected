==========================================
reduce in TEST-FEATURE-OPERATION : readFeature((((< d : DataOutPort | content :
    * # b(0) > < e : DataOutPort | content : real(0) # bool(false) >) < c :
    DataInPort | content : r(0) # b(0),cache : r(0) # b(1) >) < b : DataInPort
    | content : r(0) # bool(true),cache : r(1) # bool(true) >) < a : DataInPort
    | content : r(0) # bool(false),cache : r(1) # bool(false) >) .
rewrites: 49
result FeatureReadResult: {< a : DataInPort | content : [0] # [false],cache :
    r(0) # [false] > < b : DataInPort | content : [0] # [false],cache : r(0) #
    [true] > < c : DataInPort | content : [0] # [false],cache : r(0) # b(0) > <
    d : DataOutPort | content : * # b(0) > < e : DataOutPort | content : [0] #
    [false] >,a |-> (r(1) # [false] : [false]), b |-> (r(0) # [true] : [true]),
    c |-> (r(0) # (b(0) or b(1)) : b(0)), d |-> * # [false], e |-> [0] # [
    false]}
==========================================
reduce in TEST-FEATURE-OPERATION : writeFeature((d |-> * # bool(true), e |-> r(
    1) # b(1)), (((< d : DataOutPort | content : * # b(0) > < e : DataOutPort |
    content : real(0) # bool(false) >) < c : DataInPort | content : r(0) # b(
    0),cache : r(0) # b(1) >) < b : DataInPort | content : r(0) # bool(true),
    cache : r(1) # bool(true) >) < a : DataInPort | content : r(0) # bool(
    false),cache : r(1) # bool(false) >) .
rewrites: 7
result Configuration: < a : DataInPort | content : r(0) # [false],cache : r(1)
    # [false] > < b : DataInPort | content : r(0) # [true],cache : r(1) # [
    true] > < c : DataInPort | content : r(0) # b(0),cache : r(0) # b(1) > < d
    : DataOutPort | content : * # [true] > < e : DataOutPort | content : r(1) #
    b(1) >
==========================================
reduce in TEST-DATA-OPERATION : readData(< c1 : Data | value : real(3/4) # b(
    1),features : none,subcomponents : none,properties : none,connections :
    empty > < c2 : Data | value : r(0) # bool(true),features : none,
    subcomponents : none,properties : none,connections : empty >) .
rewrites: 8
result DataValuation: c1 |=> [3/4] # b(1), c2 |=> r(0) # [true]
==========================================
reduce in TEST-DATA-OPERATION : writeData((c1 |=> real(0) # bool(true), c2 |=>
    real(1) # b(1)), < c1 : Data | value : real(3/4) # b(1),features : none,
    subcomponents : none,properties : none,connections : empty > < c2 : Data |
    value : r(0) # bool(true),features : none,subcomponents : none,properties :
    none,connections : empty >) .
rewrites: 8
result Configuration: < c1 : Data | properties : none,features : none,
    subcomponents : none,connections : empty,value : [0] # [true] > < c2 : Data
    | properties : none,features : none,subcomponents : none,connections :
    empty,value : [1] # b(1) >
Bye.
